test/video_room_test.exs

```diff
@@ -1,48 +1,233 @@
 defmodule Janus.Plugin.VideoRoomTest do
-  use ExUnit.Case
+  use ExUnit.Case, async: false
   import Mock
+  import VideoRoomTest.Helper
   alias Janus.Plugin.VideoRoom
+  alias Janus.Plugin.VideoRoom.{CreateRoomProperties, EditRoomProperties}
 
   @id 1
   @room_name "room_name"
+  @handle_id 1
 
-  describe "create_room sends create room request through connection" do
-    test "returning ok tuple with room id when call succeeds" do
+  describe "create_room/6 sends create room request through connection and" do
+    test "returns ok tuple with room id on success" do
       description = "A room description"
 
-      with_mock Janus.Connection,
-        call: fn _, message ->
+      with_mock Janus.Session,
+        execute_request: fn _, message ->
           assert message[:body][:room] == @room_name
           assert message[:body][:description] == description
           {:ok, %{"videoroom" => "created", "room" => @id}}
         end do
-        room_props = %VideoRoom{description: description}
+        room_props = %CreateRoomProperties{description: description}
 
         assert {:ok, @id} ==
-                 VideoRoom.create_room(@room_name, room_props, Janus.Conenction, 1, 1)
+                 VideoRoom.create_room(Janus.Session, @room_name, room_props, nil, nil)
       end
     end
 
-    test "returning an error when room already_exists" do
-      with_mock Janus.Connection,
-        call: fn _, _message ->
-          {:ok, %{"error_code" => 427, "videoroom" => "event"}}
+    test_videoroom_plugin_error(
+      :room_already_exists,
+      &VideoRoom.create_room/5,
+      [Janus.Session, @room_name, %CreateRoomProperties{}, nil, nil]
+    )
+
+    test_session_error_propagation(
+      &VideoRoom.create_room/5,
+      [Janus.Session, @room_name, %CreateRoomProperties{}, nil, nil]
+    )
+  end
+
+  describe "edit/5 sends edit room request through connection and" do
+    test "returns ok tuple with room id on success" do
+      with_mock Janus.Session,
```

Comment:
I'll let it slip, but in general such mocks should be avoided and it could be quite easily achieved here. You can read about it: http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/ https://hexdocs.pm/mox/Mox.html